### ISSUE TITLE
Typings issues

### ISSUE THREAD

**Issue by whitecolor** [AgustinCB/docker-api#56]
You are writing with TS, and still typing seems to be not good, even `@types/dockernode` feel to have more proper typings. 

Even your `new Docker(/*opts: not typed*/)` constructor has no typings for options, and results are typed not correctly:

![image](https://user-images.githubusercontent.com/736697/48704557-bb448400-ec18-11e8-947b-6cf6be0b7898.png)

It is not a critique, just want to draw your attention to typings issues.

**Comment by AgustinCB** [comment]
You're right. When I did this project, I was just starting with Typescript and missed a lot of things. I'd like to fix this, but I can't promise an ETA because I have other projects on the go.

That being said, contributions are welcomed.

**Comment by MartinMuzatko** [comment]
I'm also here for the search for proper types. 
It should be easy though. you could just add `@types/dockerode` as a dependency and just re-use them.

Types are somewhat up to date: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/dockerode/index.d.ts
